applications/CompressiblePotentialFlowApplication/custom_conditions/incompressible_adjoint_potential_wall_condition.cpp

```diff
@@ -0,0 +1,254 @@
+
+    pNewCondition->SetData(this->GetData());
+    pNewCondition->SetFlags(this->GetFlags());
+
+    return pNewCondition;
+}
+
+template <class TPrimalCondition>
+void AdjointIncompressiblePotentialWallCondition<TPrimalCondition>::Initialize() 
+{   
+    mpPrimalCondition->Initialize();
+}
+
+template <class TPrimalCondition>
+void AdjointIncompressiblePotentialWallCondition<TPrimalCondition>::InitializeSolutionStep(ProcessInfo& rCurrentProcessInfo) 
+{
+    mpPrimalCondition->Data() = this->Data();
+    mpPrimalCondition->Set(Flags(*this));
+    mpPrimalCondition->InitializeSolutionStep(rCurrentProcessInfo);
+}
+template <class TPrimalCondition>
+void AdjointIncompressiblePotentialWallCondition<TPrimalCondition>::GetValuesVector(Vector& rValues, int Step) 
+{
+
+    KRATOS_TRY
+
+    if(rValues.size() != TNumNodes)
+        rValues.resize(TNumNodes, false);
+
+    bool is_kutta=false;
+    for(unsigned int i=0; i<TNumNodes; i++){
+        if (GetGeometry()[i].FastGetSolutionStepValue(DISTANCE)<0.0){
+            is_kutta=true;
+            break;
+        }
+    }
+    for(unsigned int i=0; i<TNumNodes; i++){
+        if(is_kutta){
+            if(GetGeometry()[i].FastGetSolutionStepValue(DISTANCE)<0.0)
+                rValues[i] = GetGeometry()[i].FastGetSolutionStepValue(ADJOINT_VELOCITY_POTENTIAL);
+            else
+                rValues[i] = GetGeometry()[i].FastGetSolutionStepValue(ADJOINT_AUXILIARY_VELOCITY_POTENTIAL);
+        }
+        else
+            rValues[i] = GetGeometry()[i].FastGetSolutionStepValue(ADJOINT_VELOCITY_POTENTIAL);
+    }
+
+    KRATOS_CATCH("");
+
+}
+
+template <class TPrimalCondition>
+void AdjointIncompressiblePotentialWallCondition<TPrimalCondition>::CalculateLeftHandSide(MatrixType &rLeftHandSideMatrix,
+                            ProcessInfo &rCurrentProcessInfo) 
+{
+    VectorType RHS;
+    this->CalculateLocalSystem(rLeftHandSideMatrix, RHS, rCurrentProcessInfo);
+    rLeftHandSideMatrix.clear();
+}
+
+template <class TPrimalCondition>
+void AdjointIncompressiblePotentialWallCondition<TPrimalCondition>::CalculateSensitivityMatrix(const Variable<array_1d<double,3> >& rDesignVariable,
+                                        Matrix& rOutput,
+                                        const ProcessInfo& rCurrentProcessInfo) 
+{
+    if (rOutput.size1() != TNumNodes)
+        rOutput.resize(TDim*TNumNodes, TNumNodes, false);
+    rOutput.clear();
+}
+
+template <class TPrimalCondition>
+void AdjointIncompressiblePotentialWallCondition<TPrimalCondition>::CalculateLocalSystem(MatrixType &rLeftHandSideMatrix,
+                            VectorType &rRightHandSideVector,
+                            ProcessInfo &rCurrentProcessInfo) 
+{               
+    if (rLeftHandSideMatrix.size1() != TNumNodes)
+        rLeftHandSideMatrix.resize(TNumNodes, TNumNodes, false);
+    if (rRightHandSideVector.size() != TNumNodes)
+        rRightHandSideVector.resize(TNumNodes, false);
+    rLeftHandSideMatrix.clear();
```

Comment:
out of curiosity, why do you call clear here and in many other places?

Reply:
This was like this in the original element from Riccardo so we left it just to make sure they are initialized as zero.

Reply:
The condition is currently not contributing to the adjoint LHS/RHS, but it might do in further developments or in other response functions, so for now i just make sure they are all zero by calling Clear(). Also the [adjoint scheme](https://github.com/KratosMultiphysics/Kratos/blob/46c73754b9b6e95b47c6e1838b2925cc201baa13/kratos/solving_strategies/schemes/residual_based_adjoint_static_scheme.h#L192) needs the condition LocalSystem, so for now the condition just has dummy functions.

Reply:
Side question, is it necessary to correct all 'trailing whitespace' suggestions from codacy? Or can the PR go through even if the Codacy Check fails? I say this to avoid waiting for Travis / Appveyor builds just for this reason.

Reply:
2 comments:
1. Codacy is just a suggestion, it is not mandatory like Travis & Appveyor
2. Many of us have set our editors to automatically remove trailing whitespaces when saving a file, this way we never have to deal with this. E.g. see here: https://stackoverflow.com/questions/30884131/remove-trailing-spaces-automatically-or-with-a-shortcut